cmd/terramate/cli/cli.go

```diff
@@ -204,15 +185,50 @@ func newCLI(
 		return nil, fmt.Errorf("failed to parse cli args %v: %v", args, err)
 	}
 
+	wd := parsedArgs.Chdir
+	if wd == "" {
+		wd, err = os.Getwd()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get working directory: %w", err)
+		}
+	}
+
+	wd, err = filepath.EvalSymlinks(wd)
+	if err != nil {
+		return nil, fmt.Errorf("failed evaluating symlinks for %q: %w", wd, err)
+	}
+
+	err = os.Chdir(wd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to change working directory to %q: %w", wd, err)
+	}
+
+	foundRoot, prj, err := lookupProject(wd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup project root from %q: %w", wd, err)
+	}
+
+	if !foundRoot {
+		return nil, fmt.Errorf("project root not found")
+	}
```

Comment:
Maybe because of how Go map access works , but it feels more intuitive to have:

```
prj, found, err := lookupProject(wd)
```

Instead of the first return value being a boolean. Not a major (or even minor =P) issue of course, just sharing =P.